.github/workflows/macos.yml

```diff
@@ -0,0 +1,34 @@
+name: "MacOS"
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-macos:
+    name: ${{ matrix.os }} Build FreeOrion SDK
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-10.15]
+    env:
+      CMAKE_BUILD_PARALLEL_LEVEL: 2
+      HOMEBREW_NO_AUTO_UPDATE: 1
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v3
+      - name: Configure
+        run: |
+          mkdir build
+          cd build
+          cmake -G "Xcode" -D BOOST_PARALLEL_JOBS=$(sysctl hw.ncpu | awk '{print $2}') ..
+      - name: Build
+        run: |
+          cd build
+          cmake --build . --config RelWithDebInfo -- -parallelizeTargets
```

Comment:
On my system, I only run `cmake --build . --config RelWithDebInfo`, without the `-- -parallelizeTargets` part.